### ISSUE TITLE
Dependency generation results in incorrect paths

### ISSUE THREAD

**Issue by donno** [brechtsanders/winlibs_mingw#40]
### Steps to reproduce

1.  Download and extract winlibs-i686-posix-dwarf-gcc-10.2.0-mingw-w64-8.0.0-r5.7z
2. Create a simple hello world program like so called example.cpp
```c++
#include <iostream>

int main() {
 	std::cout << "Hello world!" << std::endl;
	return 0;
}
``` 
3. Compile like so `.\mingw32\bin\g++.exe -c -MD -MF example.d .\example.cpp`
4. Check example.d

### Expected results
The paths within the dependency file are valid Windows file paths.

For example like so:
```
d:\programs\development\winlibs-i686-posix-dwarf-gcc-10.2.0-mingw-w64-8.0.0-r5\mingw32\include\c++\10.2.0\iostream \
```
### Actual Result
The paths have been mangled. For example:
```
...
d\:\programs\development\winlibs-i686-posix-dwarf-gcc-10.2.0-mingw-w64-8.0.0-r5\mingw32\include\c++\10.2.0\iostream \
d\:\programs\development\winlibs-i686-posix-dwarf-gcc-10.2.0-mingw-w64-8.0.0-r5\mingw32\include\c++\10.2.0\ios \
d\:\programs\development\winlibs-i686-posix-dwarf-gcc-10.2.0-mingw-w64-8.0.0-r5\mingw32\include\c++\10.2.0\iosfwd \
....
```

(The trailing  \ on the end is part of the Make syntax for line continuation).

**Comment by brechtsanders** [comment]
I'm able to reproduce this. The backslashes must really confuse g++.
Doing the same with clang++ gives the right paths. That is they have single backslashes, but lines end with a backslash too, which looks like the newline is being escaped there, so the backslashes in the path should probable be escaped too.

Have you reported this to the gcc developers?

**Comment by donno** [comment]
I haven't reported it to the gcc developers as I wasn't sure if it was something unique to your builds. It was unclear if you had applied any patches on top trying to give it a more Windows feel.  

I've tested with your GCC 8.4.0 and 9.3.0 releases and they produce the correct paths.  

I also tested with your Win64 GCC 10.2.0 release as that was what I was originally intending to report on. That did have the problem.

So looks like it may be a regression in the 10 series.

I was also able to minimise it further by calling the preprocess (cpp) directly like so:
`.\mingw64\bin\cpp .\example.cpp -M`
